Single frontend for Lambda and Batch (#376)
aws-samples/aws-hpc-tutorials: content/05-aws-batch-and-lambda/setup/02-02-create-docker-container-index.en.md

+++
title = "2. Provision Lambda, Batch and S3 buckets, etc.."
weight = 22
+++

> In this section, you will deploy Lambda, Batch and S3 buckets with [CloudFormation](https://aws.amazon.com/cloudformation/) templates to provision infrastructure as code, so the system can be managed and scaled to multi-region easily.

Reviewer: ```suggestion
In this section, you will deploy Lambda, Batch and S3 buckets with [AWS CloudFormation](https://aws.amazon.com/cloudformation/) templates to provision infrastructure as code, so the system can be managed and scaled to multi-region easily.
```

> The Bash script "buildArch.sh" can be found at the top level of the "fsi-demo" directory. You can start to build the infrastructure with the following command:

Reviewer: ```suggestion
You will start to build the infrastructure with the following command:
```

Using collected information with the template file ```CloudFormation/fsi-demo-batch.yaml```, the script will deploy AWS Batch, including creating a compute environment, a job queue, a job definition and related IAM roles. Batch can support multiple clients and very large-scale workloads with additional Batch queues and compute environments.

Then the script will deploy the remaining infrastructures with ```CloudFormation/fsi-demo-s3.yaml```, such as input and output S3 buckets, lambda function and its S3 trigger, EventBridge rule for Batch jobs triggered from S3 events etc..

```
$ cat buildArch.sh 
#!/bin/bash

> 

Reviewer: Too complicated.
